### PR TITLE
[IMP] website_sale_hide_price: hide prices at website level

### PR DESCRIPTION
**Pull request by amkarthik**
This PR will add support to hide price at website level.

cc: @SodexisTeam

### CODE REVIEW

File: website_sale_hide_price/data/product_snippet_template_data.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
         <xpath expr="//span[hasclass('fw-bold')]" position="before">
             <div
                 id="website_hide_price"
-                t-if="user_id.partner_id.website_show_price and not data.get('website_hide_price')"
+                t-if="not website.website_hide_price and user_id.partner_id.website_show_price and not data.get('website_hide_price')"
```
Comment (chienandalu) [inline website_sale_hide_price/data/product_snippet_template_data.xml]: suggestion: You just need to consider the new field in the `_compute_website_show_price` method, so it has precedence over the env.user flag

**Review by chienandalu — CHANGES_REQUESTED** [review]

File: website_sale_hide_price/data/product_snippet_template_data.xml (hunk 1 above)
Comment (amkarthik) [reply website_sale_hide_price/data/product_snippet_template_data.xml]: @chienandalu 
Thanks for the suggestion.
Modified it accordingly

File: website_sale_hide_price/models/website.py [hunk 2]
```diff
@@ -18,4 +23,8 @@ class Website(models.Model):
 
     def _compute_website_show_price(self):
         for rec in self:
-            rec.website_show_price = request.env.user.partner_id.website_show_price
+            rec.website_show_price = (
+                False
+                if rec.website_hide_price
+                else request.env.user.partner_id.website_show_price
+            )
```
Comment (chienandalu) [inline website_sale_hide_price/models/website.py]: suggestion: easier to read:
```suggestion
            rec.website_show_price = not rec.website_hide_price or request.env.user.partner_id.website_show_price
```

File: website_sale_hide_price/views/res_config_settings_views.xml [hunk 3]
```diff
@@ -11,14 +11,17 @@
                     title="Set a default text when hiding prices"
                 >
                     <div class="o_setting_left_pane">
-
+                        <field name="website_hide_price" />
```
Comment (chienandalu) [inline website_sale_hide_price/views/res_config_settings_views.xml:14]: This is confusing. Put this setting in its own setting box

File: website_sale_hide_price/views/website_sale_template.xml [hunk 4]
```diff
@@ -32,16 +34,21 @@
...
+                add="website,website.website_show_price,product.website_hide_price"
+                separator=","
+            />
         </xpath>
         <xpath expr="//a[@id='add_to_cart']" position="before">
             <t t-set="user_authenticated" t-value="user_id != website.user_id" />
             <div
                 class="alert alert-info"
-                t-if="product.website_hide_price and (product.website_hide_price_message or website.website_hide_price_default_message) and user_authenticated"
+                t-if="(not website.website_show_price or product.website_hide_price or not user_id.sudo().partner_id.website_show_price) and (product.website_hide_price_message or website.website_hide_price_default_message) and user_authenticated"
```
Comment (chienandalu) [inline website_sale_hide_price/views/website_sale_template.xml]: This isn't meant to be triggered by the general setting. The aim is that independently of the general setting we can either show or not custom messages on selected products, but we could hide thos messages without deleting the message itself.

**Review by chienandalu — CHANGES_REQUESTED** [review]
Thanks, much simpler :smile: 

Some comments:

File: website_sale_hide_price/models/website.py (hunk 2 above)
Comment (amkarthik) [reply website_sale_hide_price/models/website.py]: Modified it.
Thanks

File: website_sale_hide_price/views/res_config_settings_views.xml (hunk 3 above)
Comment (amkarthik) [reply website_sale_hide_price/views/res_config_settings_views.xml:14]: Moved it to its own div.

File: website_sale_hide_price/views/website_sale_template.xml (hunk 4 above)
Comment (amkarthik) [reply website_sale_hide_price/views/website_sale_template.xml]: Reverted it to as it was before.
Thanks.

**Comment by amkarthik** [comment]
@chienandalu 
Done with the suggested changes.
Thanks!

File: website_sale_hide_price/views/res_config_settings_views.xml [hunk 5]
```diff
@@ -8,10 +8,22 @@
...
                     class="col-12 col-lg-6 o_setting_box"
                     id="website_sale_hide_price"
-                    title="Set a default text when hiding prices"
+                    title="Hide prices at website level"
                 >
                     <div class="o_setting_left_pane">
+                        <field name="website_hide_price" />
+                    </div>
+                    <div class="o_setting_right_pane">
+                        <label for="website_hide_price" />
```
Comment (chienandalu) [inline website_sale_hide_price/views/res_config_settings_views.xml:17]: suggestion: add a hint
```suggestion
                        <label for="website_hide_price" />
                        <div class="text-muted">
                            Hide the product prices for everyone only on this website
                        </div>
```

Comment (amkarthik) [reply website_sale_hide_price/views/res_config_settings_views.xml:17]: Looks good, Updated it
Thanks.

File: website_sale_hide_price/models/website.py [hunk 6]
```diff
@@ -18,4 +23,7 @@ class Website(models.Model):
 
     def _compute_website_show_price(self):
         for rec in self:
-            rec.website_show_price = request.env.user.partner_id.website_show_price
+            rec.website_show_price = (
+                not rec.website_hide_price
+                or request.env.user.partner_id.website_show_price
```
Comment (chienandalu) [inline website_sale_hide_price/models/website.py]: This should be an `and` (my bad)
```suggestion
                and request.env.user.partner_id.website_show_price
```

Comment (amkarthik) [reply website_sale_hide_price/models/website.py]: Fixed it.
Thanks

**Review by chienandalu — APPROVED** [review]
Great. Now it's working :+1: 

please check @pilarvargas-tecnativa

**Review by pilarvargas-tecnativa — APPROVED** [review]
thanks :)

**Pull request merged**